shared/src/main/scala/fr/hmil/roshttp/tools/io/IO.scala

```diff
@@ -65,13 +65,17 @@ private[roshttp] object IO {
 
   /** Reads the entire content of an input stream as a byte array. */
   def readInputStreamToByteArray(stream: InputStream): Array[Byte] = {
+    require(stream != null, "stream must not be null")
```

Comment:
Since we are working in Scala, null values are prohibited everywhere. Option should be used whenever we want pass either something or nothing.
There is no argument for checking those values in particular when in fact no value should ever be null in the whole codebase.